vllm/v1/attention/backends/flex_attention.py

```diff
@@ -250,28 +335,66 @@ def __post_init__(self):
         assert self.cu_prefix_query_lens is None, "Not implemented yet."
         assert self.prefix_kv_lens is None, "Not implemented yet."
         assert self.suffix_kv_lens is None, "Not implemented yet."
+        self.doc_ids = _offsets_to_doc_ids_tensor(self.query_start_loc)
         self.num_blocks = self.total_cache_tokens // self.block_size
         self.mask_mod = self.get_mask_mod()
-        self.block_mask = self.build_block_mask()
+        # NB: This is a very slow way of building this block mask
```

Comment:
I want to refactor this to be a `custom` block-mask builder that is the default and doesn't go through create_block_mask